### DISCUSSION TITLE
issue compiling single header

### DISCUSSION THREAD

**Discussion by skreamz** [skypjack/entt#821] [Q&A]
Hi I'm using the single header of entt with vc2022 using the 'Preview - Features from the Latest C++ Working Draft (/std:c++latest) 'language standard, and I've the following errors when compiling.

Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0526	a parameter may not have void type	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	15290	
Error (active)	E0018	expected a ')'	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	15290	
Error	C2143	syntax error: missing ')' before '*=' (compiling source file src\components\Component.cpp)	Engine	D:\Projects\ Engine\libraries\entt\entt.hpp	13116	
Error	C2143	syntax error: missing ';' before '*=' (compiling source file src\components\Component.cpp)	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	13116	
Error	C2059	syntax error: '*=' (compiling source file src\components\Component.cpp)	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	13116	
Error	C2059	syntax error: ')' (compiling source file src\components\Component.cpp)	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	13116	
Error	C2334	unexpected token(s) preceding '{'; skipping apparent function body (compiling source file src\components\Component.cpp)	Engine	D:\Projects\Engine\libraries\entt\entt.hpp	13116	

and a whole load more of the same. the first 2 are for the following.

template<typename Ret, typename... Args>
	delegate(Ret(*)(const void *, Args...), const void *= nullptr)->delegate<Ret(Args...)>;

any help would be grateful.
Thank you.

**Comment by skypjack** [comment]
I don't have a working copy of VS2022 at the moment but this looks a lot like the C++ version to use doesn't propagate to the imported library.
How do you include EnTT in your codebase? A VS solution? CMake? What else?

**Reply by skreamz** [reply]
VS solution via addition include directories option.  I have a single include file called headers.h which include all libraries including entt.hpp, I've also tried it with a new/blank project, same errors.

**Reply by skypjack** [reply]
Is your project online? I can clone it when I'm back at work and let you know what's wrong.
